lib/cartopy/mpl/ticker.py

```diff
@@ -99,12 +192,15 @@ def _hemisphere(self, value, value_source_crs):
 class LatitudeFormatter(_PlateCarreeFormatter):
     """Tick formatter for latitude axes."""
     def __init__(self, degree_symbol=u'\u00B0', number_format='g',
-                 transform_precision=1e-8):
+                 transform_precision=1e-8, decimal=False,
```

Comment:
I'm concerned that the kwarg `decimal=False` introduces a breaking change.  In your tests for these Formatters you have added an extra kwarg `decimal=True` to test that the result remains the same as it was previously.

In reality, that means that if we keep the syntax of those tests the same, the result will now be different, which is a breaking change.  That means that we can only release these changes for v1.0.0, which is the next major release of Cartopy.

To address this, you can either flip the default value of the `decimal` kwarg so that the behaviour is the same by default, or you can create a v1.0.x branch and target that instead of master.  Your call.

Reply:
I think it is important that `decimal=False`  by default for everybody working at small areas. However, it remains less important than the capability of setting it manually.
So, is the good solution to let it at `True` by default until v1.0.0, then switch to `False`?

This highlights another need: to extend the list of keyword accepted by `Gridliner.__init__` for tuning  everything. In this particular case, a keyword like `xformatter_kwargs` would be useful to tune the default formatter at creation time: `xformatter_kwargs:{'decimal': False}`. By extension, all the tuning that is accessible by changing attributes like `xlines`or `xpadding` should be accessible from the keywords. Should we add the keywords and in which order?

And by the way, v1.0.0 must be the occasion to replace `x/y` by `m/p` when appropriate since X/Y no longer necessarily means longitude/latitude but meridian/parallel.

Reply:
In the examples you make a reference to, degrees are displayed as fractions of degree, despite values are plain degrees with a fraction equals to zero. This means that for a smaller area, labels would be like ``35.5°W``. If we want to keep this behavior by default, decimal must be set to True in the formatter classes (and locators) since ``decimal=False`` means no minutes/seconds.
In one of the last commits, I added a decimal kwargs also to ``add_gridliner`` and ``Gridliner`` to easily change this important behavior.

Reply:
Actually in those examples, the top one shows the default behaviour of gridline labels (as demonstrated by not using the kwarg `decimal` at all) as not decimal.

The lower example has a decimal point but this is spec ified by the kwarg `number_format` (which does make me wonder which would override if both `decimal=False` and `number_format=.1f` were used).

Reply:
And now that I think about that a little more, I think it is a bad idea to be able to set 2 kwargs which can contradict each other; this is asking for trouble.

I understand the requirement to be able to use whole numbers only in Latitude and Longitude Formatters, but we can get all the control we need over it with the kwarg `number_format` can't we?

In which case, I would suggest removing `decimal` completely.

Thoughts?

Reply:
The ``number_format`` is simply ignored when ``decimal=False``.
By letting the decimal keyword, we allow people working in small area to access the hability to have minutes and seconds in their labels without too much effort.

In think there is no contradiction. However, it may be clearer to use another keyword name that decimal.

Reply:
You're right, I misunderstood the output format when `decimal=True`.

To me decimal implies a number divided into 10, whereas you need to describe a number divided into 60.  How about a boolean arg for `minutes`?

Reply:
Yep. ``minutes`` is probably less confusing than ``decimal``. So, when ``minutes=True``, the locators will be able to fall on fractions of 60, the formatter will display minutes and seconds when appropriate, and ``number_format`` will be deactivated.
So I rename it everywhere?

Reply:
Yes please, and set it to `False` as default so that we don't change the default behaviour. Are you happy with that?

Reply:
But what is does is produce [Decimal degrees](https://en.wikipedia.org/wiki/Decimal_degrees). `minutes` is not right because it can also produces seconds.

Reply:
I quite like `dms`.  I'd be happy with that.

Reply:
I also agree. So minutes->dms?

Reply:
Yes please!